client/components/forms/form-currency-input/index.tsx

```diff
@@ -0,0 +1,91 @@
+import './style.scss';
+import { Gridicon } from '@automattic/components';
+import clsx from 'clsx';
+import { ChangeEvent } from 'react';
+import FormSelect from 'calypso/components/forms/form-select';
+import FormTextInputWithAffixes from 'calypso/components/forms/form-text-input-with-affixes';
+
+interface FormCurrencyInputProps extends React.SelectHTMLAttributes< HTMLInputElement > {
+	className?: string;
+	currencyList?: Currency[];
+	currencySymbolPrefix: string;
+	currencySymbolSuffix?: string;
+	id: string;
+	name: string;
+	noWrap?: boolean;
+	placeholder: string;
+	value?: number;
+
+	onCurrencyChange?: ( event: ChangeEvent< HTMLSelectElement > ) => void;
+}
+
+interface Currency {
+	code: string;
+	label?: string;
+}
+
+export default function FormCurrencyInput( props: FormCurrencyInputProps ): JSX.Element {
+	const {
+		className,
+		currencySymbolPrefix,
+		currencySymbolSuffix,
+		onCurrencyChange,
+		currencyList,
+		placeholder = '0.00',
+		...inputProps
+	} = props;
+
+	return (
+		<FormTextInputWithAffixes
+			{ ...inputProps }
+			className={ clsx( 'form-currency-input', className ) }
+			prefix={ renderAffix( currencySymbolPrefix, onCurrencyChange, currencyList ) }
+			suffix={ renderAffix( currencySymbolSuffix, onCurrencyChange, currencyList ) }
+			type="number"
+			step={ currencySymbolPrefix === 'JPY' ? '1' : '0.01' }
+			placeholder={ placeholder }
+			value={ props.value || null }
+		/>
+	);
+}
+
+function renderAffix(
+	currencySymbol: string | undefined,
+	onCurrencyChange: ( ( event: ChangeEvent< HTMLSelectElement > ) => void ) | undefined,
+	currencyList: Currency[] | undefined
+): JSX.Element | null {
+	// If the currency value is not defined, don't render this affix at all
+	if ( ! currencySymbol ) {
```

Comment:
minor Q, but why did we rename currencyValue to currencySymbol? And it seems the comments still refer to it as a value. Is it the value or something else?

Reply:
Fixed [84e391d](https://github.com/Automattic/wp-calypso/pull/102895/commits/84e391d21fa5c7a99092060472296184dd45a81c)

Reply:
that makes perfect sense to me, thanks!